### DISCUSSION TITLE
A few questions about the capabilities of OpenROAD-Flow-Scripts

### DISCUSSION THREAD

**Discussion by lokki11199** [The-OpenROAD-Project/OpenROAD#3447] [Q&A] [answered]
Hello! I have a few questions about ORFS capabilities.

1)	Does OpenROAD support libraries with CCS driver model?
2)	Is incremental flow supported?
3)	Does OpenSTA in OpenROAD support all sdc commands?
4)	Is SystemVerilog support limited to the constructs that Yosys supports?
5)	Does OpenROAD do only DRC analysis or also DRV and DFM? Of the DFM analysis I have only seen an antenna rule check.
6)	Does OpenROAD provide corner-based optimisation, or it can only show us the characteristics, like wns, for different corners?
7)	Are OCVs supported?
8)	Does OpenROAD support crosstalk delay/noise analysis?

Thank you for your answers!

**Comment by maliberty** [comment, marked as answer]
> Hello! I have a few questions about ORFS capabilities.
> 
> 1. Does OpenROAD support libraries with CCS driver model?

Only NLDM is currently supported but CCS is under development.

> 2. Is incremental flow supported?

That can have a lot of meanings.  We do support restarting the flow at a given step.

> 3. Does OpenSTA in OpenROAD support all sdc commands?

Yes afaik but be careful that there are many proprietary commands that are not in SDC.

> 4. Is SystemVerilog support limited to the constructs that Yosys supports?

Yes.  The is the surelog plugin that has a good amount of SV support

> 5. Does OpenROAD do only DRC analysis or also DRV and DFM? Of the DFM analysis I have only seen an antenna rule check.

OR does max slew/cap/fanout if that's what you mean by DRV.  There is no DFM.

> 6. Does OpenROAD provide corner-based optimisation, or it can only show us the characteristics, like wns, for different corners?

OR can be setup for multi-corner analysis.  ORFS doesn't take advantage of that today.

> 7. Are OCVs supported?

Yes but see 6

> 8. Does OpenROAD support crosstalk delay/noise analysis?

No.  Once we have CCS for delay it would be natural to tackle this next.

> 
> Thank you for your answers!

**Reply by vvbandeira** [reply]
@dralabeing. Should we curate this and ask Jack to add it to OR and ORFS FAQs?

**Reply by vijayank88** [reply]
I am updating ORFS tutorial. So I'll update ORFS FAQ's.
Let Jack to update OR FAQ's.

**Comment by dralabeing** [comment]
> @dralabeing. Should we curate this and ask Jack to add it to OR and ORFS FAQs?

Yes, this is a good idea. Let's add this to the FAQs
